### ISSUE TITLE
AS3933 Functionality Testing Error

### ISSUE THREAD

**Issue by IndieMash** [LieBtrau/arduino-as3933#13]
Hi LieBtrau,

I trust this message finds you well.

Allow me to express my utmost respect and admiration for your expertise and contributions in the field. Having followed your work for a considerable period, I am deeply impressed by your knowledge and dedication.

Turning to the matter at hand, I must admit my relatively novice status in the realm of electronics. Regrettably, I have not yet purchased an oscilloscope for diagnostic purposes. I humbly request your patience and understanding in this regard.

I have recently undertaken the assembly of an AS3933 module and employed your script to conduct testing. However, I encountered an error message during the process, which I have outlined below:

02:30:11.587 -> test1: Checking AS3933 functionality
02:30:11.638 -> Tuning antenna: LF1
02:30:11.638 -> Testing bit: 4
02:30:12.616 -> Frequency: 1
02:30:12.616 -> Testing bit: 3
02:30:13.615 -> Frequency: 0
02:30:13.615 -> Testing bit: 2
02:30:14.610 -> Frequency: 0
02:30:14.610 -> Testing bit: 1
02:30:15.608 -> Frequency: 0
02:30:15.608 -> Testing bit: 0
02:30:16.606 -> Frequency: 0
02:30:16.606 -> Calibration for antenna 1 fails

For this endeavor, I utilized an array of equipment, including an Arduino Nano [Atmel atmega 328pb], a TXS0108E Voltage-Level Shifter, a CAS143-47 [3D Antenna], and a self-assembled AS3933 module.

I am seeking guidance on potential issues with the schematic, which I have attached for your reference. Additionally, I am uncertain whether modifications to the code are necessary to ensure compatibility with the Arduino Nano [Atmel atmega 328pb Xplained mini].

Furthermore, I relied upon the inductance values specified in the CAS143-47 datasheet [4.7mH & 7.2mH] to calculate the necessary capacitance for resonance at the desired frequency of 125kHz. The calculated values stand at 345pF for the 4.7mH (X, Y) antennas and 225pF for the 7.2mH (Z) antenna.

Regrettably, due to the unavailability of the precise capacitors required for this project, I resorted to utilizing multiple capacitors in parallel. Could this substitution potentially contribute to the encountered error?

Moreover, the AS3933 datasheet underscores the significance of utilizing a resistor with a value of either 1k, 3k, 9k, or 27k Ohms, contingent upon the application. Despite adhering to this directive with a resistor value of 9k Ohms, I am apprehensive that this selection may be inadequate, as several examples I have reviewed suggest the use of resistors exceeding 200k Ohms. If 9k Ohms proves unsuitable, I would greatly appreciate guidance on the recommended resistor value to ensure optimal performance.

In conclusion, I am eager to receive your expert insights and recommendations on potential adjustments or improvements to rectify the encountered issue. Your enlightenment would be valuable to my ongoing project endeavors.
[schm.pdf](https://github.com/LieBtrau/arduino-as3933/files/14904294/schm.pdf)

**Comment by LieBtrau** [comment]
Hi IndieMash,

You probably noticed that this repository hasn't received an update in 8 years.  My interest has shifted to other projects.  Over the years several people (likely students looking for an easy project) tried this project and failed.  It looks easy because it's only low frequency and requires only I²C or SPI, but don't let that fool you.
If you don't have access to an oscilloscope, it will be very hard to get things right.  Chatgpt is good at writing text but not at fixing issues like this.

1. On https://www.sciosense.com/as3933-lf-receiver-ic/ you can find more download material and probably reference design for this IC.  
2. Your calculation for 345pF looks correct.  I haven't checked the other one.
3. Check the additional info to know more about the damping resistor value.  The datasheet mentions that an **internal** damping resistor from 1k up to 27k can be added in parallel to the external damping resistor.  Your external damping resistor should be much larger than 27k so that it causes less damping.  200k for the external damping resistor might be a good value.
4. As for your schematic, you don't need to place so many parallel capacitors.  Two caps in parallel to each inductor should be enough.  The AS3933 has internal caps (up to 31pF) that will allow you to fine tune the resonance frequency.  There's really no point in placing the 1pF and 1.2pF.  So for the 345pF, you can just put 330pF and then use the internal register to add internal caps until the resonance frequency is ok.  For the 225pF, you can just put a 220pF capacitor and tune then with the internal caps.
5. It's misleading how you've drawn the parallel RLC-networks.  You'd better adjust your schematic symbol so that you can draw the RLC-network as fig.29 on the datasheet.

Regards,

Christoph

**Comment by IndieMash** [comment]
Hi Christoph, 

Thank you for the quick response.

I will implement your advice today and get back to you as soon as possible.

I would also like to find out whether there will be any modification required for the script to work well on an Arduino Nano or is the script provided good enough? 

Regards
Malcolm
